wordpress-mobile/WordPress-Android
# Glide - comments

Fixes #7962 

Replaces Volley with Glide in the `Comments` and `Reader - Comments` sections. 
Adds support for loading animated GIF images in both sections.

Adds new subclass of Html.ImageGetter (`WPCustomImageGetter`), which takes care of replacing `<img>` tags with actual images using Glide. Glide supports loading resources into ImageViews or so called Targets. Since we display comments into a TextView using Html.fromHtml() method, we must implement a custom Target (`WPRemoteResourceViewTarget`) with support for loading multiple images into a single TextView.  `WPRemoteResourceViewTarget` also supports displaying placeholder and replacing it when the request finishes with remote/error image. More info about Targets can be found [here](http://bumptech.github.io/glide/doc/targets.html) and [here](https://bumptech.github.io/glide/doc/resourcereuse.html#attempting-to-load-two-different-resources-into-the-same-target).

Since this is quite a significant change and it's easy to miss something, can you please both review it. Thanks!!

To test:
1. Post a comment with 2+ gif images (using web)
2. Go to My Site -> Comments
3. Make sure avatars look as expected
4. Open detail of the comment
5. Make sure avatar looks as expected
6. Notice loading placeholder is being displayed
7. Notice all images are animated
8. Click on the name of the blog in the header
9. Click on comments in the bottom bar
10. Notice all images are animated
![gif-comments2](https://user-images.githubusercontent.com/2261188/42751870-770a8c6a-88ec-11e8-9482-28c9f1e4e31e.gif)
